torch/_dynamo/eval_frame.py

```diff
@@ -409,16 +418,7 @@ def __init__(
         compiler_config=None,
     ):
         def on_enter():
-            global most_recent_backend
-            if OptimizeContext._different_backend(most_recent_backend, compiler_fn):
-                if config.raise_on_backend_change:
-                    raise ResetRequired()
-                else:
-                    warnings.warn(
-                        "changing options to `torch.compile()` may require "
-                        "calling `torch._dynamo.reset()` to take effect"
-                    )
-            most_recent_backend = compiler_fn
+            _set_current_backend(compiler_fn)
```

Comment:
I don't think this will work correctly with nesting.  We may need to store the prior backend and restore it on exit.